File: pythonAlgorithm/Practice/1823找出游戏的获胜者.py
class Solution:
    def findTheWinner(self, n: int, k: int) -> int:
        q = [i for i in range(1, n+1)]
        cur, tn = 0, n
        while tn > 1:
            cur = (cur + k - 1) % tn
            q.pop(cur)
            tn -= 1

        return q[0]

File: Go_LC/模拟/1823找出游戏的获胜者.go
package main

// 约瑟夫问题，迭代法
func findTheWinner(n int, k int) int {
    if n == 1{
        return 1
    }
    winner := 1
    for i := 2; i <= n; i ++{
        winner = (winner + k - 1) % i + 1
    }

    return winner
}